crates/core/tedge_agent/src/operation_workflows/actor.rs

```diff
@@ -465,6 +482,71 @@ impl WorkflowActor {
             _ => Err(CommandTopicError::InvalidCommandTopic),
         }
     }
+
+    fn process_iterate(
+        &mut self,
+        state: GenericCommandState,
+        target_json_path: String,
+        handlers: IterateHandlers,
+    ) -> Result<GenericCommandState, IterationError> {
```

Comment:
I would try to move this logic to `tedge_api`, the goal being to keep a clear separation between workflow definition and workflow execution.

Reply:
Resolved by 1012f41c0069b302fb54aa249149fdee9ab20071